### PR TITLE
Feature/api design 1.0

### PR DESCRIPTION
**Pull request by SamRagusa**
One thing that should probably still be added is setting attributes from a pandas dataframe.

### CODE REVIEW

**Review by StokesMIDE — CHANGES_REQUESTED** [review]
The only real request for a change is the bit about using `self.domain`; the rest are mostly comments and suggestions.

File: endaq/cloud/core.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+        """
+        response = requests.get(self.domain + "/api/v1/files",
+                                params={'limit': limit},
+                                headers={"x-api-key": self.api_key})
+
+        files_json_data = response.json()['data']
+
+        devices = {}
+        for f_data in files_json_data:
+            if f_data['device'] is not None and len(f_data['device']) > 0:
```
[the remark this answers is not in the document]
Comment (SamRagusa) [reply endaq/cloud/core.py:167]: This doesn't work, if `f_data['device']` is None, then what you're suggesting will evaluate to True, but I need it to evaluate to False in that case.   (I've also tested it and it fails in that case)

File: endaq/cloud/core.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
+"""
+Core enDAQ Cloud communication API
+
+TODO: Rewrite docstrings (pref. before implementing methods)
```
Comment (StokesMIDE) [inline endaq/cloud/core.py:4]: Just delete this line and I think it's good to go.

**Review by StokesMIDE — CHANGES_REQUESTED** [review]
One minor change and all's good.

File: endaq/cloud/core.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (CrepeGoat) [reply endaq/cloud/core.py:174]: @StokesMIDE correct me if I'm wrong, but I think this is what you're getting at:
```pycon
>>> def v1(obj):
...     return (obj is not None and len(obj) > 0)
...
>>> def v2(obj):
...     return bool(obj)
...
>>> o = None
>>> assert v1(o) == v2(o)
>>> o = []
>>> assert v1(o) == v2(o)
>>> o = ["non-empty list"]
>>> assert v1(o) == v2(o)
```

In that case, I think you mean `if f_data['device']`

**Pull request merged**